packages/resolve-dependencies/src/index.ts

```diff
@@ -173,7 +174,15 @@ export default async function (
       }
     }
   }
-  const { newLockfile, pendingRequiresBuilds } = updateLockfile(dependenciesGraph, opts.wantedLockfile, opts.virtualStoreDir, opts.registries) // eslint-disable-line:prefer-const
+
+  // If --fix-lockfile is specified, the previous lockfile's packages is ignored
+  // and the packages field(all field execept version) will be generated from the new resolved depNode
+  const { newLockfile, pendingRequiresBuilds } = updateLockfile(dependenciesGraph, opts.fixLockfile
```

Comment:
Probably `fixLockfile` is not needed here. Just use `opts.forceFullResolution`.

Reply:
I think so. I don't want to affect existing logic so a new opion was passed, if forceResolution is better I will use force resolution and cover it with test

Reply:
if the existing tests will pass with forceFullResolution, then it is OK